PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,26 @@
+### IMPORTANT: Please do not create a Pull Request without creating an issue first.
+
+### Title
+Please remove this section and [use conventional commits structure](https://www.conventionalcommits.org/en/v1.0.0/) for PR titles.  
+E.g. feat(api): add Foo to Bar
+
+### Description
+A one to two sentence summary of the PR for easy digestion, must be different and more descriptive than title. Combination of title and description should summarize the issue + solution.
```

Comment:
Can we merge Description and Changes?

Will this get filled in automatically from the long commit message?

Reply:
A description is distinctly different from the list of changes.

A list of changes dives into implementation details while a description keeps the higher level goals of the PR at the front of mind and provides a good summary without getting bogged down in details.

Reply:
Isn't the list of changes just the list of commits?

Reply:
sure, and if we're going to couple those concepts in our workflow and then enforce a conventional commits structure to our commits, we don't even need this section.

but from my past experience and current observations of commit messages in gcn so far, commits (names and descriptions) may not be truly reflective of their changes or may oversimplify many changes into a description, so this section is necessary to give the developer time to pause and organize their work and thoughts in a systematic way that reflects with brevity and clarity what was done as part of the PR in a detail oriented fashion.

a change list also allows us to confirm that the changes here are truly the only changes in the PR, more words and cross-references serve to improve overall code quality and simplify the review process for reviewers.

Try it, it might feel like a pointless time sink at first, but I assure you it's for improving everyones ergonomics in the long term.

Reply:
We have not inculcated a tradition of detailed commit messages. I think we should do that --- instead of listing out changes in the PRs.

Reply:
Clearly this is coupled with establishing a commit message style guide.

Reply:
Not necessarily. That is simply your interpretation. This is intentionally not coupled to commit messages to give us greater flexibility and provide momentum in the short term. We cannot tackle all of the things all at once. This PR with templates gets us one step closer in the right direction to establishing a typical SWE process.